### PR TITLE
Improved code-quality for ts-add

### PR DESCRIPTION
**Pull request by Whleee**
Changed createEditedStudent to use clone.
Abstracted isAvailableAtDateTime and hasAttendance checks into a separate method isStudentAvailable.

### CODE REVIEW

**Review by faqihisaninja — APPROVED** [review]
LGTM

File: src/main/java/seedu/canoe/logic/commands/AddStudentToTrainingCommand.java [hunk 1]
```diff
@@ -171,11 +167,27 @@ public boolean uniqueChecker(Training trainingToCheck, Student check) throws Com
...
+
     /**
      * Returns the Student object in the model with the Id same as the specified Unique Id.
      * @param model
      * @param id
-     * @return
+     * @return the Student Object that corresponds to the specified Id.
      */
     public Student getStudentWithID(Model model, String id) throws CommandException {
         id.trim();
```
Comment (sudogene) [inline src/main/java/seedu/canoe/logic/commands/AddStudentToTrainingCommand.java:193]: Should be `id = id.trim()`
`trim()` returns a new String, does not modify the old String.

**Review by sudogene — APPROVED** [review]
LGTM

File: src/main/java/seedu/canoe/logic/commands/AddStudentToTrainingCommand.java (hunk 1 above)
Comment (yejiadong) [reply src/main/java/seedu/canoe/logic/commands/AddStudentToTrainingCommand.java:193]: fixed

**Pull request merged**